Sources/NIOCore/NIOLoopBound.swift

```diff
@@ -107,6 +107,26 @@ public final class NIOLoopBoundBox<Value>: @unchecked Sendable {
         return .init(_value: nil, uncheckedEventLoop: eventLoop)
     }
 
+    /// Initialise a ``NIOLoopBoundBox`` by sending a ``Sendable`` value, validly callable off `eventLoop`.
+    ///
+    /// Contrary to ``init(_:eventLoop:)``, this method can be called off `eventLoop` because we know that `value` is ``Sendable``.
+    /// So we don't need to protect `value` itself, we just need to protect the ``NIOLoopBoundBox`` against mutations which we do because the ``value``
+    /// accessors are checking that we're on `eventLoop`.
+    public static func makeBoxSendingValue(
```

Comment:
I find it weird to use a factory method here personally. I would prefer using an init here. I also see little value in having this constrained to `Sendable` since `Sendable` types can freely be shared between threads.

I expected a spelling like this

```
init(value: sending Value, eventLoop: any EventLoop) {}
```

This would allow moving an non-Sendable value into an EL's isolation region which is dynamically enforced by the LoopBoundBox.

Reply:
- factory is to get "more words", the `init`s are unconstrained but have to run on `eventLoop`
- `sending` isn't ready yet (#2754)
- there's a huge value for `Sendable` values. It's not about sending the _values_ around, `NIOLoopBoundBox` is about _mutating_ the values. Typically this is for thread-safe network clients/servers that have a state machine which only gets accessed on a particular EL. So current: `var state: State // only access on self.eventLoop`; new: `let state: NIOLoopBound<State>`

Reply:
That's not accurate.

- `class Box<T> { var value: T }` cannot be Sendable (without synchronisation), you get _mutation_ and _sharing_.
- `class NIOLoopBoundBox<T> { var value: T}` is always `Sendable` because it checks that all accesses are on the same EL. So you get _mutation_ but no _sharing_

Reply:
I've got no objection to the PR in its current state, but I'm not going to approve because I'd like to see @FranzBusch and @weissi get aligned here.